File: leetcode/is_subsequence.cpp
class Solution {
public:
    bool isSubsequence(string s, string t) {
        int counter = 0;
        for (int i = 0; i < t.length(); i++){
            if (t[i] == s[counter]){
                counter++;
            }
        }
        if (counter == s.length()) return true;
        return false;
    }
};

File: leetcode/golang/is_subsequence.go
func isSubsequence(s string, t string) bool {
    var pts int
    if s == "" { return true }
    for ptt := 0; ptt < len(t); ptt++ {
        if t[ptt] == s[pts] { 
            if pts == len(s) - 1 { return true } 
            pts++ 
        }
    }
    return false
}
